test/sqlite/result_helper.cpp

```diff
@@ -523,4 +531,26 @@ bool TestResultHelper::CompareValues(SQLLogicTestLogger &logger, MaterializedQue
 	return true;
 }
 
+bool TestResultHelper::CompareValues(SQLLogicTestLogger &logger, MaterializedQueryResult &result, string rvalue_str) {
+	bool want_match = StringUtil::StartsWith(rvalue_str, "<REGEX>:");
+	string regex_str = StringUtil::Replace(rvalue_str, "<REGEX>:", "");
```

Comment:
seems like this code is being duplicated? Can you pull it out of the other `CompareValues()` function and have that function call this one? 

Can you also rename the functions so they do not have the same name?

Reply:
Pulled out duplicating code into a separate function called MatchesResult. Now for the queries it is called from the CompareValues(), for the statements from the CheckStatementResult.